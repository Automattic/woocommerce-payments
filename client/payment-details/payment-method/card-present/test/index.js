/* eslint-disable camelcase */
/** @format */
/**
 * External dependencies
 */
import { render } from '@testing-library/react';
/**
 * Internal dependencies
 */
import CardPresentDetails from '../';

describe( 'CardPresentDetails', () => {
	beforeEach( () => {
		jest.clearAllMocks();
		global.wcSettings = {
			countries: {
				US: 'United States of America',
			},
		};
		global.wcpaySettings = {
			zeroDecimalCurrencies: [],
		};
	} );

	afterEach( () => {
		delete global.wcSettings;
		delete global.wcpaySettings;
	} );

	test( 'renders loading', () => {
		const charge = {
			payment_method_details: {
				type: 'card_present',
				card_present: {
					last4: '9999',
					fingerprint: '123456789abc',
					exp_month: '11',
					exp_year: '2023',
					funding: 'funding',
					network: 'network',
					country: 'US',
				},
			},
			billing_details: {
				name: 'foo',
				email: 'bar',
				formattedAddress: 'baz',
			},
		};

		const { container } = render(
			<CardPresentDetails charge={ charge } isLoading={ false } />
		);

		expect( container ).toMatchSnapshot();
	} );
} );
