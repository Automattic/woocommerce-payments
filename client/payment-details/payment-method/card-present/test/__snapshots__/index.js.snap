// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPresentDetails renders loading 1`] = `
<div>
  <div
    class="payment-method-details"
  >
    <div
      class="payment-method-details__column"
    >
      <div
        class="payment-method-detail"
      >
        <h4
          class="payment-method-detail__label"
        >
          Number
        </h4>
        <p
          class="payment-method-detail__value"
        >
          •••• 
          9999
        </p>
      </div>
      <div
        class="payment-method-detail"
      >
        <h4
          class="payment-method-detail__label"
        >
          Expires
        </h4>
        <p
          class="payment-method-detail__value"
        >
          11 / 2023
        </p>
      </div>
      <div
        class="payment-method-detail"
      >
        <h4
          class="payment-method-detail__label"
        >
          Type
        </h4>
        <p
          class="payment-method-detail__value"
        >
          Network undefined card
        </p>
      </div>
      <div
        class="payment-method-detail"
      >
        <h4
          class="payment-method-detail__label"
        >
          ID
        </h4>
        <p
          class="payment-method-detail__value"
        />
      </div>
    </div>
    <div
      class="payment-method-details__column"
    >
      <div
        class="payment-method-detail"
      >
        <h4
          class="payment-method-detail__label"
        >
          Owner
        </h4>
        <p
          class="payment-method-detail__value"
        >
          foo
        </p>
      </div>
      <div
        class="payment-method-detail"
      >
        <h4
          class="payment-method-detail__label"
        >
          Owner email
        </h4>
        <p
          class="payment-method-detail__value"
        >
          bar
        </p>
      </div>
      <div
        class="payment-method-detail"
      >
        <h4
          class="payment-method-detail__label"
        >
          Address
        </h4>
        <p
          class="payment-method-detail__value"
        >
          <span />
        </p>
      </div>
      <div
        class="payment-method-detail"
      >
        <h4
          class="payment-method-detail__label"
        >
          Origin
        </h4>
        <p
          class="payment-method-detail__value"
        >
          United States of America
        </p>
      </div>
    </div>
  </div>
</div>
`;
